s3/s3.go

```diff
@@ -35,9 +35,13 @@ func New(bucket, prefix, region string, timeoutSeconds int, creds *ecs.Credentia
 	}
 }
 
-func (s3 *S3) Put(name string, payload []byte, contentType string, rw http.ResponseWriter) ([]byte, error) {
+func (s3 *S3) request(httpMethod string, name string, payload []byte, contentType string, rw http.ResponseWriter) ([]byte, error) {
 	uri := s3.bucketUri + s3.prefix + "/" + name
-	req, err := http.NewRequest("PUT", uri, bytes.NewReader(payload))
+	var payloadReader io.Reader = nil
+	if payload != nil {
+		payloadReader = bytes.NewReader(payload)
```

Comment:
Given that the payload may be big, it may be more efficient to use `bufio.NewReader(..)`